packages/core/src/fhirpath/utils.ts

```diff
@@ -105,15 +105,19 @@ function getTypedPropertyValueWithSchema(
     resultValue = input.value[path];
     resultType = types[0].code;
   } else {
+    // The path parameter can be in both "value[x]" form and "valueBoolean" form.
+    // So we need to use the element path to find the type.
+    const elementPath = (element.path.split('.').pop() as string).replace('[x]', '');
```

Comment:
When is the path parameter in either form?

If that's the case, should the usage of `path` elsewhere in this function be replaced to stay consistent?

Reply:
Good call - there was indeed a bug with the "primitive extensions" logic.  Fixed now.  Thanks